 # Quality Control (QC) of Illumina reads
 
 ---
 ## 00.Software and resources.
 + FastQC ver. 0.11.3 (Andrews 2010)
 + Cutadapt ver. 1.16 (Martin 2011)
 + Prinseq (Schmieder & Edwards 2011).

 
 ---
 ## 01.FastQC
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/00_RawData

 for species in Dato Dbrb DkoeA DkoeB; do
  cd ${WD}/${species}
  mkdir ${WD}/01_QC/${species}
  fastqc --threads 8 --outdir ${WD}/01_QC/${species}/${species}.fastqc ${species}.PE.R1.fq.gz ${species}.PE.R2.fq.gz
 ~~~~


 ## 02.Cutadapt & Prinseq
 ~~~~
 #!/bin/bash
 
 RD=/home/nmoreyra/cbuzz/00_RawData
 WD=/home/nmoreyra/cbuzz/02_Trimming

 for species in Dato Dbrb DkoeA DkoeB; do
  cd ${WD}/${species}
  cutadapt -b file:${WD}/adapters/adapters-PE.fasta -B file:${WD}/adapters/adapters-PE.fasta --minimum-length 25 -o ${WD}/${species}/${species}.PE.cutadapt.R1.fq -p ${WD}/${species}/${species}.PE.cutadapt.R2.fq ${RD}/${species}/${species}.PE.R2.fq.gz ${RD}/${species}/${species}.PE.R2.fq.gz

  prinseq-lite.pl -verbose -fastq ${WD}/${species}/${species}.PE.cutadapt.R1.fq -fastq2 ${WD}/${species}/${species}.PE.cutadapt.R1.fq -out_good ${species}.PE.cutadapt.prinseq -out_bad ${species}.bad_prinseq -log ${species}.log -min_len 20 -trim_left 5 -min_qual_score 25
 done
 ~~~~
 

---
 # References
 + Schmieder, R., & Edwards, R. (2011). Quality control and preprocessing of metagenomic datasets. Bioinformatics, 27(6), 863-864.
 + Martin, M. (2011). Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet. journal, 17(1), 10-12.
 + Schmieder, R., & Edwards, R. (2011). Quality control and preprocessing of metagenomic datasets. Bioinformatics, 27(6), 863-864.
