 # *de novo* Genome Assembly
 
 The genomes of *D. antonietae*, *D. borborema* and both *D. koepferae* strains were sequenced following a hybrid approach that involved short and long reads sequencing technologies. First, we sequenced Illumina paired-end reads for the four genomes. Second, Pacific Biosciences (hereafter PacBio) long reads were sequenced in two stages. Initially, we sequenced the genomes of *D. borborema* and *D. koepferae* A using the RS II technology and, subsequently, the four genomes genomes were sequenced using the Sequel I technology.
 To assemble the four genomes we followed a de novo hybrid approach adapted from [Jaworski, Allan, and Matzkin. (2020)](https://matzkinlab.files.wordpress.com/2020/06/jaworski-et-al-2020.pdf). The complete scheme of this protocol can be found in [Figure S1](imgs/FigureS1_Assembly-protocol.png).

 ## 01.Low heterozygosity genome assembly using Illumina paired-end reads
 ~~~~
 #!/bin/bash
 WD=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus
 READS=/home/nmoreyra/cbuzz/02_Trimming

 for species in Dato Dbrb DkoeA DkoeB; do
  cd ${WD}/${species}
  
  platanus assemble -o ${species}.ass -f ${READS}/${species}.PE.cutadapt.prinseq.R[12].fq -k 31 -t 32 -m 120 2> ${species}.ass.log
 done
 ~~~~

 ## 02.Assembly using PacBio long reads

 ### 02.01.Canu assembly
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB;
  #correct reads
  mkdir -p ${WD}/${species}/correct && cd ${WD}/${species}/correct
  canu -correct -p ${species} -d . genomeSize=160m -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=100gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q windfall" -pacbio-raw ${READS}/${species}/${species}.pacbio.fa.gz

  #trim
  mkdir -p ${WD}/${species}/trim && cd ${WD}/${species}/trim
  canu -trim -p ${species} -d . genomeSize=160m -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=100gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q windfall" -pacbio-corrected ${WD}/${species}/correct/${species}.correctedReads.fasta.gz

  #assemble
  mkdir -p ${WD}/${species}/assembly && cd ${WD}/${species}/assembly
  canu -assemble -p ${species} -d . genomeSize=160m correctedErrorRate=0.075 -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=90gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q standard" -pacbio-corrected ${WD}/${species}/trim/${species}.trimmedReads.fasta.gz
 done
 ~~~~

 ### 02.02.Arrow polishing
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  cd ${WD}/${species}/Arrow
  REF=/home/nmoreyra/cbuzz/03_Assembly/02_Canu/${species}/assembly

  pbalign --nproc 28 ${READS}/${species}.PacBio.bam $REF/${species}.contigs.fasta ${WD}/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.contigs.fasta

  arrow -j 28 -r $REF/${species}.contigs.fasta -o ${WD}/${species}.arrow_polished.fa ${WD}/${WD}/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 02.03.Pilon polishing
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/Illumina
 
 for species in Dato Dbrb DkoeA DkoeB;
  mkdir -p ${WD}/${species}/Pilon && cd ${WD}/${species}/Pilon
  ALIGNDIR=${WD}/${species}/Pilon/Alignment
  mkdir -p $ALIGNDIR
  REF=${WD}/${species}/Arrow

  bowtie2-build ${REF}/${species}.arrow_polished.fa ${REF}/${species}.arrow_polished.fa

  bowtie2 --threads 28 -X 550 -x ${REF}/${species}.arrow_polished.fa -1 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R1.fq -2 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R2.fq -S ${ALIGNDIR}/IlluminaPE_to_arrow.sam

  samtools view -Sb $ALIGNDIR/IlluminaPE_to_arrow.sam > $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools sort -o $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools index $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam
  ###rm -f $ALIGNDIR/IlluminaPE_to_arrow.sam $ALIGNDIR/IlluminaPE_to_arrow.bam 

  java -Xmx110G -jar ~/soft/pilon-1.22.jar --genome ${REF}/${species}.arrow_polished.fa --frags $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam --output ${species}.pilon_polished --outdir ${WD}/${species} --threads 28
 done
 ~~~~

 ## 03.Hybrid assembly using Illumina and PacBio reads

 ### 03.01.DBG2OLC assembly
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC
 CONTIGS=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus/
 PBREADS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Assembly && cd ${WD}/${species}/Assembly
  DBG2OLC k 17 KmerCovTh 2 MinOverlap 30 AdaptiveTh 0.01 LD 0 MinLen 200 Contigs $CONTIGS/${species}/${species}.ass_contig.fa RemoveChimera 1 f $PBREADS/${species}/${species}.pacbio.fa.gz
 done
 ~~~~

 ### 03.02.Arrow polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  mkdir -p ${WD}/${species}/Arrow && cd ${WD}/${species}/Arrow
  REF=${WD}/${species}/Assembly

  pbalign --nproc 28 ${READS}/${species}/${species}.PacBio.bam $REF/${species}.DBG2OLC.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.DBG2OLC.fa

  arrow -j 28 -r $REF/${species}.DBG2OLC.fa -o ${WD}/${species}/Arrow/${species}.arrow_polished.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 03.03.Pilon polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC
 READS=/home/nmoreyra/cbuzz/00_RawData/Illumina

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Pilon && cd ${WD}/${species}/Pilon
  ALIGNDIR=${WD}/${species}/Pilon/Alignment
  mkdir -p $ALIGNDIR
  REF=${WD}/${species}/Arrow

  bowtie2-build ${REF}/${species}.arrow_polished.fa ${REF}/${species}.arrow_polished.fa

  bowtie2 --threads 28 -X 550 -x ${REF}/${species}.arrow_polished.fa -1 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R1.fq -2 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R2.fq -S ${ALIGNDIR}/IlluminaPE_to_arrow.sam

  samtools view -Sb $ALIGNDIR/IlluminaPE_to_arrow.sam > $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools sort -o $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools index $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam
  ###rm -f $ALIGNDIR/IlluminaPE_to_arrow.sam $ALIGNDIR/IlluminaPE_to_arrow.bam 

  java -Xmx110G -jar ~/soft/pilon-1.22.jar --genome ${REF}/${species}.arrow_polished.fa --frags $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam --output ${species}.pilon_polished --outdir ${WD}/${species} --threads 28
 done
 ~~~~

 ## 04.Merge: merging PacBio and hybrid assemblies

 ### 04.01.Quickmerge
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/04_Merge

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Quickmerge && cd ${WD}/${species}/Quickmerge
  PACBIO=/home/nmoreyra/cbuzz/03_Assembly/02_Canu/${species}/Pilon/${species}.pilon_polished.fa
  HYBRID=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC/${species}/Pilon/${species}.pilon_polished.fa

  nucmer -l 100 --prefix merge $HYBRID $PACBIO
  delta-filter -i 95 -r -q merge.delta > merge.rq.delta
  quickmerge -d merge.rq.delta -q $PACBIO -r $HYBRID -hco 5.0 -c 1.5 -l 4000000 -ml 5000
  #l change based on N50 value of each assembly
 done
 ~~~~

 ### 04.02.Arrow polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/04_Merge
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  mkdir -p ${WD}/${species}/Arrow && cd ${WD}/${species}/Arrow
  REF=${WD}/${species}/Quickmerge

  pbalign --nproc 28 ${READS}/${species}/${species}.PacBio.bam $REF/${species}.DBG2OLC.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.merged.fa

  arrow -j 28 -r $REF/${species}.merged.fa -o ${WD}/${species}/Arrow/${species}.arrow_polished.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 04.03.Pilon polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/04_Merge
 READS=/home/nmoreyra/cbuzz/00_RawData/Illumina

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Pilon && cd ${WD}/${species}/Pilon
  ALIGNDIR=${WD}/${species}/Pilon/Alignment
  mkdir -p $ALIGNDIR
  REF=${WD}/${species}/Arrow

  bowtie2-build ${REF}/${species}.arrow_polished.fa ${REF}/${species}.arrow_polished.fa

  bowtie2 --threads 28 -X 550 -x ${REF}/${species}.arrow_polished.fa -1 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R1.fq -2 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R2.fq -S ${ALIGNDIR}/IlluminaPE_to_arrow.sam

  samtools view -Sb $ALIGNDIR/IlluminaPE_to_arrow.sam > $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools sort -o $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools index $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam
  ###rm -f $ALIGNDIR/IlluminaPE_to_arrow.sam $ALIGNDIR/IlluminaPE_to_arrow.bam 

  java -Xmx110G -jar ~/soft/pilon-1.22.jar --genome ${REF}/${species}.arrow_polished.fa --frags $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam --output ${species}.pilon_polished --outdir ${WD}/${species} --threads 28
 done
 ~~~~

 ------------------------
 # Assembly assessment: contiguity and completeness

 Assembly contiguity was assessed with Quast ver. 4.6.3 (Gurevich et al. 2013) and completeness was evaluated using BUSCO ver. 4.1.4 (Seppey, Manni, and Zdobnov 2019) for 3285 dipteran universal single-copy orthologs (BUSCO groups) obtained from OrthoDB ver. 10.1 (Kriventseva et al. 2019).

 ## Contiguity
 For each genome assembly we first splitted the scaffolds into contigs using the [fasta_split_scaffolds.py](https://github.com/enormandeau/Scripts/blob/master/fasta_split_scaffolds.py) script.

 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/04_Assembly_Assess/Contiguity
 cd ${WD}

 for sequence in contigs scaffolds; do
  ASSEMBLIES=${WD}/00_Assemblies/${sequence}

  quast --threads 8 --output-dir quast_${sequence} $ASSEMBLIES/D*.fasta
 done
 ~~~~

 ## Completeness
 ~~~~
 WD=/home/nmoreyra/cbuzz/04_Assembly_Assess/Completeness

 for species in Dari Dato Dbrb Dbuz Dhyd DkoeA DkoeB Dmel Dmex Dmoj Dnav Drep Dvir; do
  busco --in ${WD}/00_Assemblies/${species}_genome.fa --out ${species} --mode genome --lineage_dataset diptera_odb10 --cpu 8
 done
 ~~~~

 # References
 + 