 # *de novo* Genome Assembly
 
 ## 01.Platanus: low heterozygosity genome assembly
 ~~~~
 #!/bin/bash
 WD=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus
 READS=/home/nmoreyra/cbuzz/02_Trimming

 for species in Dato Dbrb DkoeA DkoeB; do
  cd ${WD}/${species}
  
  platanus assemble -o ${species}.ass -f ${READS}/${species}.PE.cutadapt.prinseq.R[12].fq -k 31 -t 32 -m 120 2> ${species}.ass.log
 done
 ~~~~

 ## 02.Canu: PacBio assembly and polishing

 ### 02.01.Canu assembly
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB;
  #correct reads
  mkdir -p ${WD}/${species}/correct && cd ${WD}/${species}/correct
  canu -correct -p ${species} -d . genomeSize=160m -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=100gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q windfall" -pacbio-raw ${READS}/${species}/${species}.pacbio.fa.gz

  #trim
  mkdir -p ${WD}/${species}/trim && cd ${WD}/${species}/trim
  canu -trim -p ${species} -d . genomeSize=160m -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=100gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q windfall" -pacbio-corrected ${WD}/${species}/correct/${species}.correctedReads.fasta.gz

  #assemble
  mkdir -p ${WD}/${species}/assembly && cd ${WD}/${species}/assembly
  canu -assemble -p ${species} -d . genomeSize=160m correctedErrorRate=0.075 -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=90gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q standard" -pacbio-corrected ${WD}/${species}/trim/${species}.trimmedReads.fasta.gz
 done
 ~~~~

 ### 02.02.Arrow polishing
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  cd ${WD}/${species}/Arrow
  REF=/home/nmoreyra/cbuzz/03_Assembly/02_Canu/${species}/assembly

  pbalign --nproc 28 ${READS}/${species}.PacBio.bam $REF/${species}.contigs.fasta ${WD}/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.contigs.fasta

  arrow -j 28 -r $REF/${species}.contigs.fasta -o ${WD}/${species}.arrow_polished.fa ${WD}/${WD}/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 02.03.Pilon polishing
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/Illumina
 
 for species in Dato Dbrb DkoeA DkoeB;
  mkdir -p ${WD}/${species}/Pilon && cd ${WD}/${species}/Pilon
  ALIGNDIR=${WD}/${species}/Pilon/Alignment
  mkdir -p $ALIGNDIR
  REF=${WD}/${species}/Arrow

  bowtie2-build ${REF}/${species}.arrow_polished.fa ${REF}/${species}.arrow_polished.fa

  bowtie2 --threads 28 -X 550 -x ${REF}/${species}.arrow_polished.fa -1 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R1.fq -2 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R2.fq -S ${ALIGNDIR}/IlluminaPE_to_arrow.sam

  samtools view -Sb $ALIGNDIR/IlluminaPE_to_arrow.sam > $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools sort -o $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools index $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam
  ###rm -f $ALIGNDIR/IlluminaPE_to_arrow.sam $ALIGNDIR/IlluminaPE_to_arrow.bam 

  java -Xmx110G -jar ~/soft/pilon-1.22.jar --genome ${REF}/${species}.arrow_polished.fa --frags $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam --output ${species}.pilon_polished --outdir ${WD/$species} --threads 28
 done
 ~~~~

 ## 03.DBG2OLC: Hybrid assembly

 ### 03.01.DBG2OLC assembly
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC
 CONTIGS=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus/
 PBREADS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Assembly && cd ${WD}/${species}/Assembly
  DBG2OLC k 17 KmerCovTh 2 MinOverlap 30 AdaptiveTh 0.01 LD 0 MinLen 200 Contigs $CONTIGS/${species}/${species}.ass_contig.fa RemoveChimera 1 f $PBREADS/${species}/${species}.pacbio.fa.gz
 done
 ~~~~

 ### 03.02.Arrow polishing
 ~~~~
 #!/bin/bash
 
 WD=
 /home/nmoreyra/cbuzz/03_Assembly/05_DBG2OLC
 CONTIGS=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus/
 PBREADS=/home/nmoreyra/cbuzz/00_RawData/PacBio
WD=/home/nmoreyra/cbuzz/03_Assembly/03_Arrow
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  cd ${WD}/${species}
  REF=/home/nmoreyra/cbuzz/03_Assembly/02_Canu/${species}/assembly

  pbalign --nproc 28 ${READS}/${species}.PacBio.bam $REF/${species}.contigs.fasta ${WD}/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.contigs.fasta

  arrow -j 28 -r $REF/${species}.contigs.fasta -o ${WD}/${species}.arrow_polished.fa ${WD}/${WD}/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 03.03.Pilon polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/05_DBG2OLC
 CONTIGS=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus/
 PBREADS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB; do
  cd $WORKDIR/${species}
  DBG2OLC k 17 KmerCovTh 2 MinOverlap 30 AdaptiveTh 0.01 LD 0 MinLen 200 Contigs $CONTIGS/${species}/${species}.ass_contig.fa RemoveChimera 1 f $PBREADS/${species}/${species}.pacbio.fa.gz
 done
 ~~~~

 ## 04.Quickmerge: merging PacBio and hybrid assemblis

 ### 04.01: Quickmerge
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/Data/07_TE-annotation/RM
 EDTA=/home/nmoreyra/Data/07_TE-annotation/EDTA #/Dbrb_nocuratedlib/Dbrb_genome.fasta
 LIBDEFAULT=nocuratedlib
 LIB="${2:-$LIBDEFAULT}"
 ~~~~

 ### 04.02: Arrow polishing
 ~~~~
 ~~~~

 ### 04.03: Pilon polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/05_DBG2OLC
 CONTIGS=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus/
 PBREADS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB; do
  cd $WORKDIR/${species}
  DBG2OLC k 17 KmerCovTh 2 MinOverlap 30 AdaptiveTh 0.01 LD 0 MinLen 200 Contigs $CONTIGS/${species}/${species}.ass_contig.fa RemoveChimera 1 f $PBREADS/${species}/${species}.pacbio.fa.gz
 done
 ~~~~