 # Genome Annotation
 
 To assemble the four genomes we followed a de novo hybrid approach adapted from [Jaworski, Allan, and Matzkin (2020)](https://matzkinlab.files.wordpress.com/2020/06/jaworski-et-al-2020.pdf). The complete scheme of this protocol can be found in [Figure S1](imgs/FigureS1_Assembly-protocol.png).

 ---
 ## 00.Software and resources.
 + MITE Hunter (https://github.com/jburnette/MITE-Hunter)
 + RepeatMasker (http://www.repeatmasker.org/RepeatMasker/)
 + [CRL Scripts ver. 1.0](scripts/CRL_Scripts1.0.tar.gz)
 + RepeatModeler ver. 1.0.11 (https://github.com/Dfam-consortium/RepeatModeler).
 + ProtExcluder ver. 1.2 (Thurmond et al. 2019; Campbell et al. 2014).
 + FlyBase release FB2019_01 (Thurmond et al. 2019).
 + UniProtKB/Swiss-Prot database release 2018_11.
 + MAKER ver. 2.31.10 (Holt and Yandell 2011)
 + SNAP ver. 2006-07-28 (Korf 2004).
 + Augustus ver. 3.2.3 ⁠(Stanke et al. 2008).
 + BUSCO ver. 3.0.2 (Simão et al. 2015).
 + eggNOG (Huerta-Cepas et al. 2019) database
 + OrthoDB ver. 9.1 (Zdobnov et al. 2017).
 + Quickmerge ver. 0.2 (Chakraborty et al. 2016).
 + Quast ver. 4.6.3 (Gurevich et al. 2013).
 + BUSCO ver. 4.1.4 (Seppey, Manni, and Zdobnov 2019).

 ---
 ## 01.Repeat Annotation

 To mask genomes before gene annotation, repetitive element identification and classification were performed following the [advanced repeat library construction tutorial of MAKER](http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/Repeat_Library_Construction-Advanced). The full protocol description can be found in [Text S3](docs/TextS3_Repeat-annotation.pdf).

 ~~~~
 #!/bin/bash
 WD=/home/nmoreyra/cbuzz/04_Annotation
 #READS=/home/nmoreyra/cbuzz/02_Trimming
 DBs=${WD}/00_Databases
 CRL=/home/nmoreyra/Data/annotation_maker/Dbo/00_repeat_masking/CRL_Scripts1.0
 
 for species in Dato Dbrb DkoeA DkoeB; do
  ASSEMBLY=${WD}/00_Assemblies/${species}
  mkdir -p ${WD}/${species}/00_Repeats && ${WD}/${species}/00_Repeats
  
  ###MITEs identification
  MITEs=${WD}/${species}/00_Repeats/MITEs
  mkdir -p ${MITEs} && cd ${MITEs}
  perl MITE_Hunter_manager.pl -i $ASSEMBLY/${species}.fa -g ${species} -c 8 -S 12345678
  cat ${species}_Step8_*.fa ${species}_Step8_singlet.fa > MITE.lib
  

  ###LTR99 identification
  LTR99=${WD}/${species}/00_Repeats/LTR99
  mkdir -p ${LTR99} && cd ${LTR99}
  gt suffixerator -db $ASSEMBLY/${species}.fa -indexname ${species} -tis -suf -lcp -des -ssp -dna
  gt ltrharvest -index ${species} -out ${species}.out99 -outinner ${species}.outinner99 -gff3 ${species}.gff99 -minlenltr 100 -maxlenltr 6000 -mindistltr 1500 -maxdistltr 25000 -mintsd 5 -maxtsd 5 -motif tgca -similar 99 -vic 10 > ${species}.result99
  gt gff3 -sort ${species}.gff99 > ${species}.gff99.sort
  gt ltrdigest -trnas ${DBs}/Dmel_tRNAs.fa ${species}.gff99.sort ${species} > ${species}.gff99.gdt
  perl ${CRL}/CRL_Step1.pl --gff ${species}.gff99.gdt
  perl ${CRL}/CRL_Step2.pl --step1 CRL_Step1_Passed_Elements.txt --repeatfile ${species}.out99 --resultfile ${species}.result99 --sequencefile $ASSEMBLY/${species}.fa --removed_repeats CRL_Step2_Passed_Elements.fasta
  #PATH here is ${WD}/${species}/00_Repeats/LTR99
  mkdir -p ${WD}/${species}/00_Repeats/LTR99/fasta_files
  mv Repeat_*.fasta ${LTR99}/fasta_files
  mv CRL_Step2_Passed_Elements.fasta ${LTR99}/fasta_files
  cd ${LTR99}/fasta_files
  perl ${CRL}/CRL_Step3.pl --directory ${LTR99}/fasta_files --step2 CRL_Step2_Passed_Elements.fasta --pidentity 60 --seq_c 25 
  mv CRL_Step3_Passed_Elements.fasta ${LTR99}
  cd ${LTR99}
  perl ${CRL}/ltr_library.pl --resultfile ${species}.result99 --step3 CRL_Step3_Passed_Elements.fasta --sequencefile $ASSEMBLY/${species}.fa
  cat ${LTR99}/lLTR_Only.lib ${MITEs}/MITE.lib > ${LTR99}/repeats_to_mask_LTR99.fasta
  RepeatMasker -lib ${LTR99}/repeats_to_mask_LTR99.fasta -nolow -dir . ${species}.outinner99
  perl ${CRL}/cleanRM.pl ${species}.outinner99.out ${species}.outinner99.masked > ${species}.outinner99.unmasked
  perl ${CRL}/rmshortinner.pl ${species}.outinner99.unmasked 50 > ${species}.outinner99.clean
  makeblastdb -in ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -dbtype prot
  blastx -query ${species}.outinner99.clean -db ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -evalue 1e-10 -num_descriptions 10 -out ${species}.outinner99.clean_blastx.out.txt
  perl ${CRL}/outinner_blastx_parse.pl --blastx ${species}.outinner99.clean_blastx.out.txt --outinner ${species}.outinner99
  perl ${CRL}/CRL_Step4.pl --step3 CRL_Step3_Passed_Elements.fasta --resultfile ${species}.result85 --innerfile passed_outinner_sequence.fasta --sequencefile ${ASSEMBLY}/${species}.fa
  makeblastdb -in lLTRs_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query lLTRs_Seq_For_BLAST.fasta -db lLTRs_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out lLTRs_Seq_For_BLAST.fasta.out
  makeblastdb -in Inner_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query Inner_Seq_For_BLAST.fasta -db Inner_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out Inner_Seq_For_BLAST.fasta.out
  perl ${CRL}/CRL_Step5.pl --LTR_blast lLTRs_Seq_For_BLAST.fasta.out --inner_blast Inner_Seq_For_BLAST.fasta.out --step3 CRL_Step3_Passed_Elements.fasta --final LTR99.lib --pcoverage 90 --pidentity 80
  
  ###LTR85 identification
  LTR85=${WD}/${species}/00_Repeats/LTR85
  mkdir -p ${LTR85} && cd ${LTR85}
  gt suffixerator -db $ASSEMBLY/${species}.fa -indexname ${species} -tis -suf -lcp -des -ssp -dna
  gt ltrharvest -index ${species} -out ${species}.out85 -outinner ${species}.outinner85 -gff3 ${species}.gff85 -minlenltr 100 -maxlenltr 6000 -mindistltr 1500 -maxdistltr 25000 -mintsd 5 -maxtsd 5 -motif tgca -similar 85 -vic 10 > ${species}.result85
  gt gff3 -sort ${species}.gff85 > ${species}.gff85.sort
  gt ltrdigest -trnas ${DBs}/Dmel_tRNAs.fa ${species}.gff85.sort ${species} > ${species}.gff85.gdt
  perl ${CRL}/CRL_Step1.pl --gff ${species}.gff85.gdt
  perl ${CRL}/CRL_Step2.pl --step1 CRL_Step1_Passed_Elements.txt --repeatfile ${species}.out85 --resultfile ${species}.result85 --sequencefile $ASSEMBLY/${species}.fa --removed_repeats CRL_Step2_Passed_Elements.fasta
  #PATH here is ${WD}/${species}/00_Repeats/LTR85
  mkdir -p ${WD}/${species}/00_Repeats/LTR85/fasta_files
  mv Repeat_*.fasta ${LTR85}/fasta_files
  mv CRL_Step2_Passed_Elements.fasta ${LTR85}/fasta_files
  cd ${LTR85}/fasta_files
  perl ${CRL}/CRL_Step3.pl --directory ${LTR85}/fasta_files --step2 CRL_Step2_Passed_Elements.fasta --pidentity 60 --seq_c 25 
  mv CRL_Step3_Passed_Elements.fasta ${LTR85}
  cd ${LTR85}
  perl ${CRL}/ltr_library.pl --resultfile ${species}.result85 --step3 CRL_Step3_Passed_Elements.fasta --sequencefile $ASSEMBLY/${species}.fa
  cat ${LTR85}/lLTR_Only.lib ${MITEs}/MITE.lib > ${LTR85}/repeats_to_mask_LTR85.fasta
  RepeatMasker -lib ${LTR85}/repeats_to_mask_LTR85.fasta -nolow -dir . ${species}.outinner85
  perl ${CRL}/cleanRM.pl ${species}.outinner85.out ${species}.outinner85.masked > ${species}.outinner85.unmasked
  perl ${CRL}/rmshortinner.pl ${species}.outinner85.unmasked 50 > ${species}.outinner85.clean
  makeblastdb -in ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -dbtype prot
  blastx -query ${species}.outinner85.clean -db ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -evalue 1e-10 -num_descriptions 10 -out ${species}.outinner85.clean_blastx.out.txt
  perl ${CRL}/outinner_blastx_parse.pl --blastx ${species}.outinner85.clean_blastx.out.txt --outinner ${species}.outinner85
  perl ${CRL}/CRL_Step4.pl --step3 CRL_Step3_Passed_Elements.fasta --resultfile ${species}.result85 --innerfile passed_outinner_sequence.fasta --sequencefile ${ASSEMBLY}/${species}.fa
  makeblastdb -in lLTRs_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query lLTRs_Seq_For_BLAST.fasta -db lLTRs_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out lLTRs_Seq_For_BLAST.fasta.out
  makeblastdb -in Inner_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query Inner_Seq_For_BLAST.fasta -db Inner_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out Inner_Seq_For_BLAST.fasta.out
  perl ${CRL}/CRL_Step5.pl --LTR_blast lLTRs_Seq_For_BLAST.fasta.out --inner_blast Inner_Seq_For_BLAST.fasta.out --step3 CRL_Step3_Passed_Elements.fasta --final LTR85.lib --pcoverage 90 --pidentity 80
  RepeatMasker -lib ${LTR99}.LTR99.lib -dir . ${LTR85}/LTR85.lib  
  
  perl ${CRL}/remove_masked_sequence.pl --masked_elements ${LTR85}/LTR85.lib.masked --outfile ${LTR85}/FinalLTR85.lib
  cat ${LTR99}/LTR99.lib ${LTR85}/FinalLTR85.lib > ${WD}/${species}/00_Repeats/allLTR.lib

  ### Collecting repetitive sequences by RepeatModeler
  RM={WD}/${species}/00_Repeats/RM
  mkdir -p ${RM} && cd ${RM}
  cat ${WD}/${species}/00_Repeats/allLTR.lib ${MITEs}/MITE.lib > ${WD}/${species}/00_Repeats/allMITE_LTR.lib
  RepeatMasker -pa 8 -lib ${WD}/${species}/00_Repeats/allMITE_LTR.lib -dir . ${ASSEMBLY}/${species}.fa
  perl ${CRL}/rmaskedpart.pl ${species}.fa.masked 50 > um.${species}.fa
  BuildDatabase -name um.${species}.db -engine ncbi um.${species}.fa
  RepeatModeler -pa 8 -database um.${species}.db
  rename 's/_[0-9]*\.[a-zA-Z]+[0-9]*//g' RM_*
  cd RM  #####ACAC QUEDE
  perl ${CRL}/repeatmodeler_parse.pl --fastafile ${RM}/RM_22412/consensi.fa.classified --unknowns repeatmodeler_unknowns.fasta --identities repeatmodeler_identities.fasta
  makeblastdb -in /home/nmoreyra/Data/annotation_maker/Dbo/00_repeat_masking/Tpases020812_db/Tpases020812DNA/Tpases020812DNA.fasta -dbtype prot
  blastx -query repeatmodeler_unknowns.fasta -db /home/nmoreyra/Data/annotation_maker/Dbo/00_repeat_masking/Tpases020812_db/Tpases020812DNA/Tpases020812DNA.fasta -evalue 1e-10 -num_descriptions 10 -out modelerunknown_blast_results.txt
  perl /home/nmoreyra/Data/annotation_maker/Dbo/00_repeat_masking/CRL_Scripts1.0/transposon_blast_parse.pl --blastx modelerunknown_blast_results.txt --modelerunknown repeatmodeler_unknowns.fasta
  mv unknown_elements.txt ModelerUnknown.lib
  cat identified_elements.txt repeatmodeler_identities.fasta > ModelerID.lib
  cd ..

 
 
 
 done
 ~~~~

 ## 02.Assembly using PacBio long reads

 ### 02.01.Canu assembly
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB;
  #correct reads
  mkdir -p ${WD}/${species}/correct && cd ${WD}/${species}/correct
  canu -correct -p ${species} -d . genomeSize=160m -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=100gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q windfall" -pacbio-raw ${READS}/${species}/${species}.pacbio.fa.gz

  #trim
  mkdir -p ${WD}/${species}/trim && cd ${WD}/${species}/trim
  canu -trim -p ${species} -d . genomeSize=160m -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=100gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q windfall" -pacbio-corrected ${WD}/${species}/correct/${species}.correctedReads.fasta.gz

  #assemble
  mkdir -p ${WD}/${species}/assembly && cd ${WD}/${species}/assembly
  canu -assemble -p ${species} -d . genomeSize=160m correctedErrorRate=0.075 -useGrid=true -maxThreads=16 -maxMemory=90 -gridEngineThreadsOption="-l select=1:ncpus=16:mem=90gb" -gridEngineMemoryOption="-l walltime=02:00:00" -gridOptions="-W group_list=cactophilic -q standard" -pacbio-corrected ${WD}/${species}/trim/${species}.trimmedReads.fasta.gz
 done
 ~~~~

 ### 02.02.Arrow polishing
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  cd ${WD}/${species}/Arrow
  REF=/home/nmoreyra/cbuzz/03_Assembly/02_Canu/${species}/assembly

  pbalign --nproc 28 ${READS}/${species}.PacBio.bam $REF/${species}.contigs.fasta ${WD}/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.contigs.fasta

  arrow -j 28 -r $REF/${species}.contigs.fasta -o ${WD}/${species}.arrow_polished.fa ${WD}/${WD}/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 02.03.Pilon polishing
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/02_Canu
 READS=/home/nmoreyra/cbuzz/00_RawData/Illumina
 
 for species in Dato Dbrb DkoeA DkoeB;
  mkdir -p ${WD}/${species}/Pilon && cd ${WD}/${species}/Pilon
  ALIGNDIR=${WD}/${species}/Pilon/Alignment
  mkdir -p $ALIGNDIR
  REF=${WD}/${species}/Arrow

  bowtie2-build ${REF}/${species}.arrow_polished.fa ${REF}/${species}.arrow_polished.fa

  bowtie2 --threads 28 -X 550 -x ${REF}/${species}.arrow_polished.fa -1 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R1.fq -2 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R2.fq -S ${ALIGNDIR}/IlluminaPE_to_arrow.sam

  samtools view -Sb $ALIGNDIR/IlluminaPE_to_arrow.sam > $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools sort -o $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools index $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam
  ###rm -f $ALIGNDIR/IlluminaPE_to_arrow.sam $ALIGNDIR/IlluminaPE_to_arrow.bam 

  java -Xmx110G -jar ~/soft/pilon-1.22.jar --genome ${REF}/${species}.arrow_polished.fa --frags $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam --output ${species}.pilon_polished --outdir ${WD}/${species} --threads 28
 done
 ~~~~

 ## 03.Hybrid assembly using Illumina and PacBio reads

 ### 03.01.DBG2OLC assembly
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC
 CONTIGS=/home/nmoreyra/cbuzz/03_Assembly/01_Platanus/
 PBREADS=/home/nmoreyra/cbuzz/00_RawData/PacBio

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Assembly && cd ${WD}/${species}/Assembly
  DBG2OLC k 17 KmerCovTh 2 MinOverlap 30 AdaptiveTh 0.01 LD 0 MinLen 200 Contigs $CONTIGS/${species}/${species}.ass_contig.fa RemoveChimera 1 f $PBREADS/${species}/${species}.pacbio.fa.gz
 done
 ~~~~

 ### 03.02.Arrow polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  mkdir -p ${WD}/${species}/Arrow && cd ${WD}/${species}/Arrow
  REF=${WD}/${species}/Assembly

  pbalign --nproc 28 ${READS}/${species}/${species}.PacBio.bam $REF/${species}.DBG2OLC.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.DBG2OLC.fa

  arrow -j 28 -r $REF/${species}.DBG2OLC.fa -o ${WD}/${species}/Arrow/${species}.arrow_polished.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 03.03.Pilon polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC
 READS=/home/nmoreyra/cbuzz/00_RawData/Illumina

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Pilon && cd ${WD}/${species}/Pilon
  ALIGNDIR=${WD}/${species}/Pilon/Alignment
  mkdir -p $ALIGNDIR
  REF=${WD}/${species}/Arrow

  bowtie2-build ${REF}/${species}.arrow_polished.fa ${REF}/${species}.arrow_polished.fa

  bowtie2 --threads 28 -X 550 -x ${REF}/${species}.arrow_polished.fa -1 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R1.fq -2 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R2.fq -S ${ALIGNDIR}/IlluminaPE_to_arrow.sam

  samtools view -Sb $ALIGNDIR/IlluminaPE_to_arrow.sam > $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools sort -o $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools index $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam
  ###rm -f $ALIGNDIR/IlluminaPE_to_arrow.sam $ALIGNDIR/IlluminaPE_to_arrow.bam 

  java -Xmx110G -jar ~/soft/pilon-1.22.jar --genome ${REF}/${species}.arrow_polished.fa --frags $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam --output ${species}.pilon_polished --outdir ${WD}/${species} --threads 28
 done
 ~~~~

 ## 04.Merge: merging PacBio and hybrid assemblies

 ### 04.01.Quickmerge
 ~~~~
 #!/bin/bash

 WD=/home/nmoreyra/cbuzz/03_Assembly/04_Merge

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Quickmerge && cd ${WD}/${species}/Quickmerge
  PACBIO=/home/nmoreyra/cbuzz/03_Assembly/02_Canu/${species}/Pilon/${species}.pilon_polished.fa
  HYBRID=/home/nmoreyra/cbuzz/03_Assembly/03_DBG2OLC/${species}/Pilon/${species}.pilon_polished.fa

  nucmer -l 100 --prefix merge $HYBRID $PACBIO
  delta-filter -i 95 -r -q merge.delta > merge.rq.delta
  quickmerge -d merge.rq.delta -q $PACBIO -r $HYBRID -hco 5.0 -c 1.5 -l 4000000 -ml 5000
  #l change based on N50 value of each assembly
 done
 ~~~~

 ### 04.02.Arrow polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/04_Merge
 READS=/home/nmoreyra/cbuzz/00_RawData/PacBio
  
 for species in Dato Dbrb DkoeA DkoeB;
  mkdir -p ${WD}/${species}/Arrow && cd ${WD}/${species}/Arrow
  REF=${WD}/${species}/Quickmerge

  pbalign --nproc 28 ${READS}/${species}/${species}.PacBio.bam $REF/${species}.DBG2OLC.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  pbindex ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam

  samtools faidx $REF/${species}.merged.fa

  arrow -j 28 -r $REF/${species}.merged.fa -o ${WD}/${species}/Arrow/${species}.arrow_polished.fa ${WD}/${species}/Arrow/${species}.aligned_PBreads.bam
 done
 ~~~~

 ### 04.03.Pilon polishing
 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/03_Assembly/04_Merge
 READS=/home/nmoreyra/cbuzz/00_RawData/Illumina

 for species in Dato Dbrb DkoeA DkoeB; do
  mkdir -p ${WD}/${species}/Pilon && cd ${WD}/${species}/Pilon
  ALIGNDIR=${WD}/${species}/Pilon/Alignment
  mkdir -p $ALIGNDIR
  REF=${WD}/${species}/Arrow

  bowtie2-build ${REF}/${species}.arrow_polished.fa ${REF}/${species}.arrow_polished.fa

  bowtie2 --threads 28 -X 550 -x ${REF}/${species}.arrow_polished.fa -1 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R1.fq -2 ${READS}/${species}/${species}.PE.cutadapt.prinseq.R2.fq -S ${ALIGNDIR}/IlluminaPE_to_arrow.sam

  samtools view -Sb $ALIGNDIR/IlluminaPE_to_arrow.sam > $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools sort -o $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam $ALIGNDIR/IlluminaPE_to_arrow.bam
  samtools index $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam
  ###rm -f $ALIGNDIR/IlluminaPE_to_arrow.sam $ALIGNDIR/IlluminaPE_to_arrow.bam 

  java -Xmx110G -jar ~/soft/pilon-1.22.jar --genome ${REF}/${species}.arrow_polished.fa --frags $ALIGNDIR/IlluminaPE_to_arrow.sorted.bam --output ${species}.pilon_polished --outdir ${WD}/${species} --threads 28
 done
 ~~~~

 ---
 # Assembly assessment: contiguity and completeness

 Assembly contiguity was assessed with Quast ver. 4.6.3 (Gurevich et al. 2013) and completeness was evaluated using BUSCO ver. 4.1.4 (Seppey, Manni, and Zdobnov 2019) for 3285 dipteran universal single-copy orthologs (BUSCO groups) obtained from OrthoDB ver. 10.1 (Kriventseva et al. 2019).

 ## Contiguity
 For each genome assembly we first splitted the scaffolds into contigs using the [fasta_split_scaffolds.py](https://github.com/enormandeau/Scripts/blob/master/fasta_split_scaffolds.py) script.

 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/04_Assembly_Assess/Contiguity
 cd ${WD}

 for sequence in contigs scaffolds; do
  ASSEMBLIES=${WD}/00_Assemblies/${sequence}

  quast --threads 8 --output-dir quast_${sequence} $ASSEMBLIES/D*.fasta
 done
 ~~~~

 ## Completeness
 ~~~~
 WD=/home/nmoreyra/cbuzz/04_Assembly_Assess/Completeness

 for species in Dari Dato Dbrb Dbuz Dhyd DkoeA DkoeB Dmel Dmex Dmoj Dnav Drep Dvir; do
  busco --in ${WD}/00_Assemblies/${species}_genome.fa --out ${species} --mode genome --lineage_dataset diptera_odb10 --cpu 8
 done
 ~~~~

 ---
 # References
 + Chakraborty, M., Baldwin-Brown, J. G., Long, A. D., & Emerson, J. J. (2016). Contiguous and accurate de novo assembly of metazoan genomes with modest long read coverage. Nucleic acids research, 44(19), e147-e147.
 + Gurevich, A., Saveliev, V., Vyahhi, N., & Tesler, G. (2013). QUAST: quality assessment tool for genome assemblies. Bioinformatics, 29(8), 1072-1075.
 + Jaworski, C. C., Allan, C. W., & Matzkin, L. M. (2020). Chromosome‐level hybrid de novo genome assemblies as an attainable option for nonmodel insects. Molecular ecology resources, 20(5), 1277-1293.
 + Kajitani, R., Toshimoto, K., Noguchi, H., Toyoda, A., Ogura, Y., Okuno, M., ... & Itoh, T. (2014). Efficient de novo assembly of highly heterozygous genomes from whole-genome shotgun short reads. Genome research, 24(8), 1384-1395.
 + Koren, S., Walenz, B. P., Berlin, K., Miller, J. R., Bergman, N. H., & Phillippy, A. M. (2017). Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation. Genome research, 27(5), 722-736.
 + Kriventseva, E. V., Kuznetsov, D., Tegenfeldt, F., Manni, M., Dias, R., Simão, F. A., & Zdobnov, E. M. (2019). OrthoDB v10: sampling the diversity of animal, plant, fungal, protist, bacterial and viral genomes for evolutionary and functional annotations of orthologs. Nucleic acids research, 47(D1), D807-D811.
 + Seppey, M., Manni, M., & Zdobnov, E. M. (2019). BUSCO: assessing genome assembly and annotation completeness. In Gene prediction (pp. 227-245). Humana, New York, NY.
 + Walker, B. J., Abeel, T., Shea, T., Priest, M., Abouelliel, A., Sakthikumar, S., ... & Earl, A. M. (2014). Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement. PloS one, 9(11), e112963.
 + Ye, C., Hill, C. M., Wu, S., Ruan, J., & Ma, Z. S. (2016). DBG2OLC: efficient assembly of large genomes using long erroneous reads of the third generation sequencing technologies. Scientific reports, 6(1), 1-9.
