 # Genome Annotation
 
 To assemble the four genomes we followed a de novo hybrid approach adapted from [Jaworski, Allan, and Matzkin (2020)](https://matzkinlab.files.wordpress.com/2020/06/jaworski-et-al-2020.pdf). The complete scheme of this protocol can be found in [Figure S1](imgs/FigureS1_Assembly-protocol.png).

 ---
 ## 00.Software and resources.
 + MITE Hunter (https://github.com/jburnette/MITE-Hunter)
 + RepeatMasker (http://www.repeatmasker.org/RepeatMasker/)
 + [CRL Scripts ver. 1.0](scripts/CRL_Scripts1.0.tar.gz)
 + RepeatModeler ver. 1.0.11 (https://github.com/Dfam-consortium/RepeatModeler).
 + ProtExcluder ver. 1.2 (Campbell et al. 2014).
 + FlyBase release FB2019_01 (Thurmond et al. 2019).
 + UniProtKB/Swiss-Prot database release 2018_11.
 + MAKER ver. 2.31.10 (Holt and Yandell 2011)
 + SNAP ver. 2006-07-28 (Korf 2004).
 + Augustus ver. 3.2.3 ⁠(Stanke et al. 2008).
 + BUSCO ver. 3.0.2 (Simão et al. 2015).
 + eggNOG (Huerta-Cepas et al. 2019) database
 + OrthoDB ver. 9.1 (Zdobnov et al. 2017).
 + Quickmerge ver. 0.2 (Chakraborty et al. 2016).
 + Quast ver. 4.6.3 (Gurevich et al. 2013).
 + BUSCO ver. 4.1.4 (Seppey, Manni, and Zdobnov 2019).

 ---
 ## 01.Repeat Annotation

 To mask genomes before gene annotation, repetitive element identification and classification were performed following the [advanced repeat library construction tutorial of MAKER](http://weatherby.genetics.utah.edu/MAKER/wiki/index.php/Repeat_Library_Construction-Advanced). The full protocol description can be found in [Text S3](docs/TextS3_Repeat-annotation.pdf).

 ~~~~
 #!/bin/bash
 WD=/home/nmoreyra/cbuzz/04_Annotation
 #READS=/home/nmoreyra/cbuzz/02_Trimming
 DBs=${WD}/00_Databases
 CRL=/home/nmoreyra/Data/annotation_maker/Dbo/00_repeat_masking/CRL_Scripts1.0
 
 for species in Dato Dbrb DkoeA DkoeB; do
  ASSEMBLY=${WD}/00_Assemblies/${species}
  mkdir -p ${WD}/${species}/00_Repeats && ${WD}/${species}/00_Repeats
  
  ###MITEs identification
  MITEs=${WD}/${species}/00_Repeats/MITEs
  mkdir -p ${MITEs} && cd ${MITEs}
  perl MITE_Hunter_manager.pl -i $ASSEMBLY/${species}.fa -g ${species} -c 8 -S 12345678
  cat ${species}_Step8_*.fa ${species}_Step8_singlet.fa > MITE.lib
  

  ###LTR99 identification
  LTR99=${WD}/${species}/00_Repeats/LTR99
  mkdir -p ${LTR99} && cd ${LTR99}
  gt suffixerator -db $ASSEMBLY/${species}.fa -indexname ${species} -tis -suf -lcp -des -ssp -dna
  gt ltrharvest -index ${species} -out ${species}.out99 -outinner ${species}.outinner99 -gff3 ${species}.gff99 -minlenltr 100 -maxlenltr 6000 -mindistltr 1500 -maxdistltr 25000 -mintsd 5 -maxtsd 5 -motif tgca -similar 99 -vic 10 > ${species}.result99
  gt gff3 -sort ${species}.gff99 > ${species}.gff99.sort
  gt ltrdigest -trnas ${DBs}/Dmel_tRNAs.fa ${species}.gff99.sort ${species} > ${species}.gff99.gdt
  perl ${CRL}/CRL_Step1.pl --gff ${species}.gff99.gdt
  perl ${CRL}/CRL_Step2.pl --step1 CRL_Step1_Passed_Elements.txt --repeatfile ${species}.out99 --resultfile ${species}.result99 --sequencefile $ASSEMBLY/${species}.fa --removed_repeats CRL_Step2_Passed_Elements.fasta
  #PATH here is ${WD}/${species}/00_Repeats/LTR99
  mkdir -p ${WD}/${species}/00_Repeats/LTR99/fasta_files
  mv Repeat_*.fasta ${LTR99}/fasta_files
  mv CRL_Step2_Passed_Elements.fasta ${LTR99}/fasta_files
  cd ${LTR99}/fasta_files
  perl ${CRL}/CRL_Step3.pl --directory ${LTR99}/fasta_files --step2 CRL_Step2_Passed_Elements.fasta --pidentity 60 --seq_c 25 
  mv CRL_Step3_Passed_Elements.fasta ${LTR99}
  cd ${LTR99}
  perl ${CRL}/ltr_library.pl --resultfile ${species}.result99 --step3 CRL_Step3_Passed_Elements.fasta --sequencefile $ASSEMBLY/${species}.fa
  cat ${LTR99}/lLTR_Only.lib ${MITEs}/MITE.lib > ${LTR99}/repeats_to_mask_LTR99.fasta
  RepeatMasker -lib ${LTR99}/repeats_to_mask_LTR99.fasta -nolow -dir . ${species}.outinner99
  perl ${CRL}/cleanRM.pl ${species}.outinner99.out ${species}.outinner99.masked > ${species}.outinner99.unmasked
  perl ${CRL}/rmshortinner.pl ${species}.outinner99.unmasked 50 > ${species}.outinner99.clean
  makeblastdb -in ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -dbtype prot
  blastx -query ${species}.outinner99.clean -db ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -evalue 1e-10 -num_descriptions 10 -out ${species}.outinner99.clean_blastx.out.txt
  perl ${CRL}/outinner_blastx_parse.pl --blastx ${species}.outinner99.clean_blastx.out.txt --outinner ${species}.outinner99
  perl ${CRL}/CRL_Step4.pl --step3 CRL_Step3_Passed_Elements.fasta --resultfile ${species}.result85 --innerfile passed_outinner_sequence.fasta --sequencefile ${ASSEMBLY}/${species}.fa
  makeblastdb -in lLTRs_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query lLTRs_Seq_For_BLAST.fasta -db lLTRs_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out lLTRs_Seq_For_BLAST.fasta.out
  makeblastdb -in Inner_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query Inner_Seq_For_BLAST.fasta -db Inner_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out Inner_Seq_For_BLAST.fasta.out
  perl ${CRL}/CRL_Step5.pl --LTR_blast lLTRs_Seq_For_BLAST.fasta.out --inner_blast Inner_Seq_For_BLAST.fasta.out --step3 CRL_Step3_Passed_Elements.fasta --final LTR99.lib --pcoverage 90 --pidentity 80
  
  ###LTR85 identification
  LTR85=${WD}/${species}/00_Repeats/LTR85
  mkdir -p ${LTR85} && cd ${LTR85}
  gt suffixerator -db $ASSEMBLY/${species}.fa -indexname ${species} -tis -suf -lcp -des -ssp -dna
  gt ltrharvest -index ${species} -out ${species}.out85 -outinner ${species}.outinner85 -gff3 ${species}.gff85 -minlenltr 100 -maxlenltr 6000 -mindistltr 1500 -maxdistltr 25000 -mintsd 5 -maxtsd 5 -motif tgca -similar 85 -vic 10 > ${species}.result85
  gt gff3 -sort ${species}.gff85 > ${species}.gff85.sort
  gt ltrdigest -trnas ${DBs}/Dmel_tRNAs.fa ${species}.gff85.sort ${species} > ${species}.gff85.gdt
  perl ${CRL}/CRL_Step1.pl --gff ${species}.gff85.gdt
  perl ${CRL}/CRL_Step2.pl --step1 CRL_Step1_Passed_Elements.txt --repeatfile ${species}.out85 --resultfile ${species}.result85 --sequencefile $ASSEMBLY/${species}.fa --removed_repeats CRL_Step2_Passed_Elements.fasta
  #PATH here is ${WD}/${species}/00_Repeats/LTR85
  mkdir -p ${WD}/${species}/00_Repeats/LTR85/fasta_files
  mv Repeat_*.fasta ${LTR85}/fasta_files
  mv CRL_Step2_Passed_Elements.fasta ${LTR85}/fasta_files
  cd ${LTR85}/fasta_files
  perl ${CRL}/CRL_Step3.pl --directory ${LTR85}/fasta_files --step2 CRL_Step2_Passed_Elements.fasta --pidentity 60 --seq_c 25 
  mv CRL_Step3_Passed_Elements.fasta ${LTR85}
  cd ${LTR85}
  perl ${CRL}/ltr_library.pl --resultfile ${species}.result85 --step3 CRL_Step3_Passed_Elements.fasta --sequencefile $ASSEMBLY/${species}.fa
  cat ${LTR85}/lLTR_Only.lib ${MITEs}/MITE.lib > ${LTR85}/repeats_to_mask_LTR85.fasta
  RepeatMasker -lib ${LTR85}/repeats_to_mask_LTR85.fasta -nolow -dir . ${species}.outinner85
  perl ${CRL}/cleanRM.pl ${species}.outinner85.out ${species}.outinner85.masked > ${species}.outinner85.unmasked
  perl ${CRL}/rmshortinner.pl ${species}.outinner85.unmasked 50 > ${species}.outinner85.clean
  makeblastdb -in ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -dbtype prot
  blastx -query ${species}.outinner85.clean -db ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -evalue 1e-10 -num_descriptions 10 -out ${species}.outinner85.clean_blastx.out.txt
  perl ${CRL}/outinner_blastx_parse.pl --blastx ${species}.outinner85.clean_blastx.out.txt --outinner ${species}.outinner85
  perl ${CRL}/CRL_Step4.pl --step3 CRL_Step3_Passed_Elements.fasta --resultfile ${species}.result85 --innerfile passed_outinner_sequence.fasta --sequencefile ${ASSEMBLY}/${species}.fa
  makeblastdb -in lLTRs_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query lLTRs_Seq_For_BLAST.fasta -db lLTRs_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out lLTRs_Seq_For_BLAST.fasta.out
  makeblastdb -in Inner_Seq_For_BLAST.fasta -dbtype nucl
  blastn -query Inner_Seq_For_BLAST.fasta -db Inner_Seq_For_BLAST.fasta -evalue 1e-10 -num_descriptions 1000 -out Inner_Seq_For_BLAST.fasta.out
  perl ${CRL}/CRL_Step5.pl --LTR_blast lLTRs_Seq_For_BLAST.fasta.out --inner_blast Inner_Seq_For_BLAST.fasta.out --step3 CRL_Step3_Passed_Elements.fasta --final LTR85.lib --pcoverage 90 --pidentity 80
  RepeatMasker -lib ${LTR99}.LTR99.lib -dir . ${LTR85}/LTR85.lib  
  perl ${CRL}/remove_masked_sequence.pl --masked_elements ${LTR85}/LTR85.lib.masked --outfile ${LTR85}/FinalLTR85.lib
  cat ${LTR99}/LTR99.lib ${LTR85}/FinalLTR85.lib > ${WD}/${species}/00_Repeats/allLTR.lib

  ### Collecting repetitive sequences by RepeatModeler
  RMo=${WD}/${species}/00_Repeats/RModeler
  mkdir -p ${RMo} && cd ${RMo}
  cat ${WD}/${species}/00_Repeats/allLTR.lib ${MITEs}/MITE.lib > ${WD}/${species}/00_Repeats/allMITE_LTR.lib
  RepeatMasker -pa 8 -lib ${WD}/${species}/00_Repeats/allMITE_LTR.lib -dir . ${ASSEMBLY}/${species}.fa
  perl ${CRL}/rmaskedpart.pl ${species}.fa.masked 50 > um.${species}.fa
  BuildDatabase -name um.${species}.db -engine ncbi um.${species}.fa
  RepeatModeler -pa 8 -database um.${species}.db
  rename 's/_[0-9]*\.[a-zA-Z]+[0-9]*//g' RM_*
  cd ${RMo}/RM
  perl ${CRL}/repeatmodeler_parse.pl --fastafile consensi.fa.classified --unknowns repeatmodeler_unknowns.fasta --identities repeatmodeler_identities.fasta
  #makeblastdb -in ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -dbtype prot #already done before
  blastx -query repeatmodeler_unknowns.fasta -db ${DBs}/Tpases020812DNA_db/Tpases020812DNA.fasta -evalue 1e-10 -num_descriptions 10 -out modelerunknown_blast_results.txt
  perl ${CRL}/CRL_Scripts1.0/transposon_blast_parse.pl --blastx modelerunknown_blast_results.txt --modelerunknown repeatmodeler_unknowns.fasta
  mv unknown_elements.txt ModelerUnknown.lib
  cat identified_elements.txt repeatmodeler_identities.fasta > ModelerID.lib
  cd ../.. #path is ${WD}/${species}/00_Repeats

  #echo "Exclusion of gene fragments"
  #mkdir -p ${WD}/${species}/00_Repeats/exclusion_genefragments && cd ${WD}/${species}/00_Repeats/exclusion_genefragments
  blastx -query ${RMo}/RM/ModelerUnknown.lib -db ${DBs}/exclusion_gene-fragments_db/db_ncbi-flybase.faa -evalue 1e-10 -num_descriptions 10 -out ModelerUnknown.lib_blast_results.txt
  ProtExcluder.pl -option ModelerUnknown.lib_blast_results.txt ${RMo}/RM/ModelerUnknown.lib #Output is ModelerUnknown.libnoProtFinal

  #combining files to create the final repeat library
  cd ${WD}/${species}/00_Repeats
  cp ${RMo}/RM/Modelerunknown.lib .
  cat ${MITEs}/MITE.lib ${WD}/${species}/00_Repeats/allLTR.lib ${RMo}/RM/ModelerID.lib > KnownRepeats.lib
  cat KnownRepeats.lib Modelerunknown.lib > allRepeats.lib

 done
 ~~~~

 ## 02.Genome annotation
 For genome annotation we followed and adapted the annotation protocol availabe at [here](https://darencard.net/blog/2017-05-16-maker-genome-annotation), which was generated to annotate the Boa constrictor genome (Card et al. 2019).

 ### 02.01.Annotation using Evidence: RNA-seq and protein data

 ~~~~
 #-----Genome (these are always required)
 genome=/home/nmoreyra/Data/annotation_maker/Dbo/genome/Dbo_genome.fasta #genome sequence (fasta file or fasta embeded in GFF3 file)
 organism_type=eukaryotic #eukaryotic or prokaryotic. Default is eukaryotic

 #-----Re-annotation Using MAKER Derived GFF3
 maker_gff= #MAKER derived GFF3 file
 est_pass=0 #use ESTs in maker_gff: 1 = yes, 0 = no
 altest_pass=0 #use alternate organism ESTs in maker_gff: 1 = yes, 0 = no
 protein_pass=0 #use protein alignments in maker_gff: 1 = yes, 0 = no
 rm_pass=0 #use repeats in maker_gff: 1 = yes, 0 = no
 model_pass=0 #use gene models in maker_gff: 1 = yes, 0 = no
 pred_pass=0 #use ab-initio predictions in maker_gff: 1 = yes, 0 = no
 other_pass=0 #passthrough anyything else in maker_gff: 1 = yes, 0 = no

 #-----EST Evidence (for best results provide a file for at least one)
 est=/home/nmoreyra/Data/annotation_maker/transcriptomes/Dbo/Dbo_filtered_transcripts.fasta #set of ESTs or assembled mRNA-seq in fasta format
 altest=/home/nmoreyra/Data/annotation_maker/transcriptomes/Dan/Dan_filtered_transcripts.fasta #EST/cDNA sequence file in fasta format from an alternate organism
 est_gff= #aligned ESTs or mRNA-seq from an external GFF3 file
 altest_gff= #aligned ESTs from a closly relate species in GFF3 format

 #-----Protein Homology Evidence (for best results provide a file for at least one)
 protein=/home/nmoreyra/Data/annotation_maker/databases/dbs-for-maker_annotations/proteins_aa.fasta  #protein sequence file in fasta format (i.e. from mutiple oransisms)
 protein_gff=  #aligned protein homology evidence from an external GFF3 file

 #-----Repeat Masking (leave values blank to skip repeat masking)
 model_org=simple #arthropods #drosophila melanogaster #select a model organism for RepBase masking in RepeatMasker
 #rmlib=/home/nmoreyra/Data/annotation_maker/Dbo/00_repeat_masking/masked_repeats/allRepeats.libnoProtFinal #provide an organism specific repeat library in fasta format for RepeatMasker
 rmlib=/home/nmoreyra/Data/annotation_maker/Dbo/00_repeat_masking/Dbo_allrepeats_noprot.lib
 repeat_protein=/home/nmoreyra/Documents/Software/maker/data/te_proteins.fasta #provide a fasta file of transposable element proteins for RepeatRunner
 rm_gff= #pre-identified repeat elements from an external GFF3 file
 prok_rm=0 #forces MAKER to repeatmask prokaryotes (no reason to change this), 1 = yes, 0 = no
 softmask=1 #use soft-masking rather than hard-masking in BLAST (i.e. seg and dust filtering)
 
 #-----Gene Prediction
 snaphmm= #SNAP HMM file
 gmhmm= #GeneMark HMM file
 augustus_species= #Augustus gene prediction species model
 fgenesh_par_file= #FGENESH parameter file
 pred_gff= #ab-initio predictions from an external GFF3 file
 model_gff= #annotated gene models from an external GFF3 file (annotation pass-through)
 est2genome=1 #infer gene predictions directly from ESTs, 1 = yes, 0 = no
 protein2genome=1 #infer predictions from protein homology, 1 = yes, 0 = no
 trna=0 #find tRNAs with tRNAscan, 1 = yes, 0 = no
 snoscan_rrna= #rRNA file to have Snoscan find snoRNAs
 unmask=0 #also run ab-initio prediction programs on unmasked sequence, 1 = yes, 0 = no
 
 #-----Other Annotation Feature Types (features MAKER doesn't recognize)
 other_gff= #extra features to pass-through to final MAKER generated GFF3 file
 
 #-----External Application Behavior Options
 alt_peptide=C #amino acid used to replace non-standard amino acids in BLAST databases
 cpus=6 #max number of cpus to use in BLAST and RepeatMasker (not for MPI, leave 1 when using MPI)
 
 #-----MAKER Behavior Options
 max_dna_len=100000 #length for dividing up contigs into chunks (increases/decreases memory usage)
 min_contig=1000 #skip genome contigs below this length (under 10kb are often useless)
 
 pred_flank=200 #flank for extending evidence clusters sent to gene predictors
 pred_stats=0 #report AED and QI statistics for all predictions as well as models
 AED_threshold=1 #Maximum Annotation Edit Distance allowed (bound by 0 and 1)
 min_protein=0 #require at least this many amino acids in predicted proteins
 alt_splice=0 #Take extra steps to try and find alternative splicing, 1 = yes, 0 = no
 always_complete=0 #extra steps to force start and stop codons, 1 = yes, 0 = no
 map_forward=0 #map names and attributes forward from old GFF3 genes, 1 = yes, 0 = no
 keep_preds=0 #Concordance threshold to add unsupported gene prediction (bound by 0 and 1)
 
 split_hit=10000 #length for the splitting of hits (expected max intron size for evidence alignments)
 single_exon=0 #consider single exon EST evidence when generating annotations, 1 = yes, 0 = no
 single_length=250 #min length required for single exon ESTs if 'single_exon is enabled'
 correct_est_fusion=0 #limits use of ESTs in annotation to avoid fusion genes

 tries=2 #number of times to try a contig if there is a failure for some reason
 clean_try=0 #remove all data from previous run before retrying, 1 = yes, 0 = no
 clean_up=0 #removes theVoid directory with individual analysis files, 1 = yes, 0 = no
 TMP= #specify a directory other than the system default temporary directory for temporary files
 ~~~~

 ### 02.02.Training of ab initio gene predictors: SNAP and Augustus
 ~~~~
 #!/bin/bash

 WD=

 for species in Dato Dbrb DkoeA DkoeB;
  
 done
 ~~~~

 ### 02.03.Annotation of evidence combined with ab initio predictors.
 
 ~~~~
 #-----Genome (these are always required)
 genome=/home/nmoreyra/Data/annotation_maker/Dbo/genome/Dbo_genome.fasta #genome sequence (fasta file or fasta embeded in GFF3 file)
 organism_type=eukaryotic #eukaryotic or prokaryotic. Default is eukaryotic

 #-----Re-annotation Using MAKER Derived GFF3
 maker_gff= #MAKER derived GFF3 file
 est_pass=0 #use ESTs in maker_gff: 1 = yes, 0 = no
 altest_pass=0 #use alternate organism ESTs in maker_gff: 1 = yes, 0 = no
 protein_pass=0 #use protein alignments in maker_gff: 1 = yes, 0 = no
 rm_pass=0 #use repeats in maker_gff: 1 = yes, 0 = no
 model_pass=0 #use gene models in maker_gff: 1 = yes, 0 = no
 pred_pass=0 #use ab-initio predictions in maker_gff: 1 = yes, 0 = no
 other_pass=0 #passthrough anyything else in maker_gff: 1 = yes, 0 = no

 #-----EST Evidence (for best results provide a file for at least one)
 est= #set of ESTs or assembled mRNA-seq in fasta format
 altest= #EST/cDNA sequence file in fasta format from an alternate organism
 est_gff=/home/nmoreyra/Data/annotation_maker/Dbo/01_maker/00_evidence_annotation_rnd1/Dbo_evidence-rnd1.maker.output/Dbo_evidence-rnd1.all.maker.est2genome.gff #aligned ESTs or mRNA-seq from an external GFF3 file
 altest_gff= #aligned ESTs from a closly relate species in GFF3 format

 #-----Protein Homology Evidence (for best results provide a file for at least one)
 protein=  #protein sequence file in fasta format (i.e. from mutiple oransisms)
 protein_gff=/home/nmoreyra/Data/annotation_maker/Dbo/01_maker/00_evidence_annotation_rnd1/Dbo_evidence-rnd1.maker.output/Dbo_evidence-rnd1.all.maker.protein2genome.gff  #aligned protein homology evidence from an external GFF3 file

 #-----Repeat Masking (leave values blank to skip repeat masking)
 model_org= #arthropods #drosophila melanogaster #select a model organism for RepBase masking in RepeatMasker
 rmlib= #provide an organism specific repeat library in fasta format for RepeatMasker
 repeat_protein= #provide a fasta file of transposable element proteins for RepeatRunner
 rm_gff=/home/nmoreyra/Data/annotation_maker/Dbo/01_maker/00_evidence_annotation_rnd1/Dbo_evidence-rnd1.maker.output/Dbo_evidence-rnd1.all.maker.repeats.gff #pre-identified repeat elements from an external GFF3 file
 prok_rm=0 #forces MAKER to repeatmask prokaryotes (no reason to change this), 1 = yes, 0 = no
 softmask=1 #use soft-masking rather than hard-masking in BLAST (i.e. seg and dust filtering)

 #-----Gene Prediction
 snaphmm=/home/nmoreyra/Data/annotation_maker/Dbo/01_maker/01_training_gene_prediction_soft/00_snap/rnd1/Dbo_evidence-rnd1.zff.length50_aed0.25.hmm #SNAP HMM file
 gmhmm= #GeneMark HMM file
 augustus_species=Drosophila_borborema #Augustus gene prediction species model
 fgenesh_par_file= #FGENESH parameter file
 pred_gff= #ab-initio predictions from an external GFF3 file
 model_gff= #annotated gene models from an external GFF3 file (annotation pass-through)
 est2genome=0 #infer gene predictions directly from ESTs, 1 = yes, 0 = no
 protein2genome=0 #infer predictions from protein homology, 1 = yes, 0 = no
 trna=1 #find tRNAs with tRNAscan, 1 = yes, 0 = no
 snoscan_rrna= #rRNA file to have Snoscan find snoRNAs
 unmask=0 #also run ab-initio prediction programs on unmasked sequence, 1 = yes, 0 = no

 #-----Other Annotation Feature Types (features MAKER doesn't recognize)
 other_gff= #extra features to pass-through to final MAKER generated GFF3 file 

 #-----External Application Behavior Options
 alt_peptide=C #amino acid used to replace non-standard amino acids in BLAST databases
 cpus=8 #max number of cpus to use in BLAST and RepeatMasker (not for MPI, leave 1 when using MPI)

 #-----MAKER Behavior Options
 max_dna_len=100000 #length for dividing up contigs into chunks (increases/decreases memory usage)
 min_contig=1000 #skip genome contigs below this length (under 10kb are often useless) 

 pred_flank=200 #flank for extending evidence clusters sent to gene predictors
 pred_stats=0 #report AED and QI statistics for all predictions as well as models
 AED_threshold=1 #Maximum Annotation Edit Distance allowed (bound by 0 and 1)
 min_protein=0 #require at least this many amino acids in predicted proteins
 alt_splice=0 #Take extra steps to try and find alternative splicing, 1 = yes, 0 = no
 always_complete=0 #extra steps to force start and stop codons, 1 = yes, 0 = no
 map_forward=0 #map names and attributes forward from old GFF3 genes, 1 = yes, 0 = no
 keep_preds=0 #Concordance threshold to add unsupported gene prediction (bound by 0 and 1)
 
 split_hit=10000 #length for the splitting of hits (expected max intron size for evidence alignments)
 single_exon=0 #consider single exon EST evidence when generating annotations, 1 = yes, 0 = no
 single_length=250 #min length required for single exon ESTs if 'single_exon is enabled'
 correct_est_fusion=0 #limits use of ESTs in annotation to avoid fusion genes

 tries=2 #number of times to try a contig if there is a failure for some reason
 clean_try=0 #remove all data from previous run before retrying, 1 = yes, 0 = no
 clean_up=0 #removes theVoid directory with individual analysis files, 1 = yes, 0 = no
 TMP= #specify a directory other than the system default temporary directory for temporary files
 ~~~~

 For the reannotation of *D. buzzatii*, the parameters  were also set in the following way:
 ~~~~
 pred_gff=/home/nmoreyra/Data/annotation_maker/Dbu/genome/annotations/Dbu_annotations_st1.gff3
 ~~~~

 ---
 # Assembly assessment: contiguity and completeness

 Assembly contiguity was assessed with Quast ver. 4.6.3 (Gurevich et al. 2013) and completeness was evaluated using BUSCO ver. 4.1.4 (Seppey, Manni, and Zdobnov 2019) for 3285 dipteran universal single-copy orthologs (BUSCO groups) obtained from OrthoDB ver. 10.1 (Kriventseva et al. 2019).

 ## Contiguity
 For each genome assembly we first splitted the scaffolds into contigs using the [fasta_split_scaffolds.py](https://github.com/enormandeau/Scripts/blob/master/fasta_split_scaffolds.py) script.

 ~~~~
 #!/bin/bash
 
 WD=/home/nmoreyra/cbuzz/04_Assembly_Assess/Contiguity
 cd ${WD}

 for sequence in contigs scaffolds; do
  ASSEMBLIES=${WD}/00_Assemblies/${sequence}

  quast --threads 8 --output-dir quast_${sequence} $ASSEMBLIES/D*.fasta
 done
 ~~~~

 ## Completeness
 ~~~~
 WD=/home/nmoreyra/cbuzz/04_Assembly_Assess/Completeness

 for species in Dari Dato Dbrb Dbuz Dhyd DkoeA DkoeB Dmel Dmex Dmoj Dnav Drep Dvir; do
  busco --in ${WD}/00_Assemblies/${species}_genome.fa --out ${species} --mode genome --lineage_dataset diptera_odb10 --cpu 8
 done
 ~~~~

 ---
 # References
 + Card, D. C., Adams, R. H., Schield, D. R., Perry, B. W., Corbin, A. B., Pasquesi, G. I., ... & Castoe, T. A. (2019). Genomic basis of convergent island phenotypes in boa constrictors. Genome biology and evolution, 11(11), 3123-3143.
 
