 # Ortholog gene evolution

 ## 00.Software and resources.
 + OrthoMCL ver. 2.0.9 (Li, Stoeckert, and Roos 2003)
 + [OrthoMCL Pipeline](https://github.com/apetkau/orthomcl-pipeline)
 + NR Reference Proteomes release 1444 database available at RefSeq (Pruitt, Tatusova, and Maglott 2007)
 + BLAST
 + PAL2NAL ver. 14 (Suyama, Torrents, and Bork 2006)
 + PAML ver. 4.9 package (Yang 307 2007, 1997)
 + MAFFT ver. 7.215 (Katoh and Standley 2013)
 + trimAl ver. 1.4.rev22 (Capella-Gutiérrez, Silla-Martínez, and
241 Gabaldón 2009)
 + IQ-TREE ver. 2.0.3 (Minh et al. 2020) 
 + eggNOG database (Huerta-Cepas et al. 2019)
 + eggNOG-mapper (Huerta-Cepas et al. 2017)

 ### 03.01.Ortholog inference
 ~~~~
 #!/bin/bash
 
 done
 ~~~~

 ### 03.02.Validation of candidate TRGs
 ~~~~
 #!/bin/bash
 
 done
 ~~~~

 ### 03.03.Molecular evolution of TRGs
 ~~~~
 #!/bin/bash
 
 done
 ~~~~

 ### 03.03.TRG functional prediction

 ## 04.Merge: merging PacBio and hybrid assemblies

 ### 04.01.Quickmerge
 ~~~~
 #!/bin/bash

 done
 ~~~~

 ### 04.02.Arrow polishing
 ~~~~
 #!/bin/bash
 
 done
 ~~~~

 ### 04.03.Pilon polishing
 ~~~~
 #!/bin/bash
 
 done
 ~~~~

 ## Completeness
 ~~~~
 #!/bin/bash

 done
 ~~~~

 ---
 # References
 + Chakraborty, M., Baldwin-Brown, J. G., Long, A. D., & Emerson, J. J. (2016). Contiguous and accurate de novo assembly of metazoan genomes with modest long read coverage. Nucleic acids research, 44(19), e147-e147.
 + Gurevich, A., Saveliev, V., Vyahhi, N., & Tesler, G. (2013). QUAST: quality assessment tool for genome assemblies. Bioinformatics, 29(8), 1072-1075.
 + Jaworski, C. C., Allan, C. W., & Matzkin, L. M. (2020). Chromosome‐level hybrid de novo genome assemblies as an attainable option for nonmodel insects. Molecular ecology resources, 20(5), 1277-1293.
 + Kajitani, R., Toshimoto, K., Noguchi, H., Toyoda, A., Ogura, Y., Okuno, M., ... & Itoh, T. (2014). Efficient de novo assembly of highly heterozygous genomes from whole-genome shotgun short reads. Genome research, 24(8), 1384-1395.
 + Koren, S., Walenz, B. P., Berlin, K., Miller, J. R., Bergman, N. H., & Phillippy, A. M. (2017). Canu: scalable and accurate long-read assembly via adaptive k-mer weighting and repeat separation. Genome research, 27(5), 722-736.
 + Kriventseva, E. V., Kuznetsov, D., Tegenfeldt, F., Manni, M., Dias, R., Simão, F. A., & Zdobnov, E. M. (2019). OrthoDB v10: sampling the diversity of animal, plant, fungal, protist, bacterial and viral genomes for evolutionary and functional annotations of orthologs. Nucleic acids research, 47(D1), D807-D811.
 + Seppey, M., Manni, M., & Zdobnov, E. M. (2019). BUSCO: assessing genome assembly and annotation completeness. In Gene prediction (pp. 227-245). Humana, New York, NY.
 + Walker, B. J., Abeel, T., Shea, T., Priest, M., Abouelliel, A., Sakthikumar, S., ... & Earl, A. M. (2014). Pilon: an integrated tool for comprehensive microbial variant detection and genome assembly improvement. PloS one, 9(11), e112963.
 + Ye, C., Hill, C. M., Wu, S., Ruan, J., & Ma, Z. S. (2016). DBG2OLC: efficient assembly of large genomes using long erroneous reads of the third generation sequencing technologies. Scientific reports, 6(1), 1-9.

aaaaaaaaa
 + (Minh, Hahn, and Lanfear 2020)
